### PR TITLE
Introduce ruff execution on the tests folder #2871

### PR DESCRIPTION
**Pull request by iAndriy**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,4 @@
+RUFF_FORMATTED_FOLDERS = sanic examples scripts tests
```
Comment (ahopkins) [inline Makefile:1]: Want to add `./guide` while you're at it?

Comment (iAndriy) [reply Makefile:1]: @ahopkins thanks for suggestion, this makes sense, though in a spirit of smaller iterations, I would rather prefer to address it separate MR, which I can definitely do.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -34,15 +34,15 @@ jobs:
           - { python-version: "3.9",  tox-env: type-checking }
           - { python-version: "3.10", tox-env: type-checking }
           - { python-version: "3.11", tox-env: type-checking }
-          - { python-version: "3.8",  tox-env: py38,          max-attempts: 3 }
-          - { python-version: "3.8",  tox-env: py38-no-ext,   max-attempts: 3 }
+          - { python-version: "3.8",  tox-env: py38,          max-attempts: 1, timeout-minutes: 20}
```
Comment (iAndriy) [inline .github/workflows/tests.yml:37]: Note: This and other adjustments to tests.yml are going to be deleted once we have passing tests.

File: tests/typing/test_typing.py
Comment (iAndriy) [inline tests/typing/test_typing.py:1]: @ahopkins This specific test started to fail, but all changes are backward compatible, I do see that parametrised test has hardcoded numbers of lines.
Do you have any take on how this test shall be addressed and how we can proceed with this ?

File: Makefile (hunk 1 above)
Comment (ahopkins) [reply Makefile:1]: 👍 sounds like a plan

**Review by ahopkins — APPROVED** [review]

**Comment by iAndriy** [comment]
@ahopkins all comments resolved and thanks for your review.

**Pull request merged**